governance/permissionless-proposals.md

```diff
@@ -0,0 +1,80 @@
+## The Workflow
+
+The general workflow would be:
+
+1. **Draft Proposal:** A delegate starts by creating a Draft proposal and posting it on the optimism governance forum. At this stage, the proposal is open to the entire community for feedback. 
+2. **Proposal Finalization**: After the proposal has been drafted, delegates provide feedback, ask clarifying questions, and propose changes. Once the delegate incorporates this input and the proposal is refined, an approved proposer will be able to call `propose` on the `Top100DelegatesProposalValidator` contract which will return a `proposalId` , aimed to be added into the forum proposal and mark it as “FINAL”.
+3. **Move to vote via Top100DelegatesProposalValidator:** With the final proposal ready, the top 100 delegates can start signing-off on the on-chain proposal using the `proposalId`. When the proposal is ready to move to voting anyone should be able to call `moveToVote` , and if the proposal meets the necessary rules, such as having the required top 100 delegates sign-offs, being submitted within the correct time window, not going over the distribution rate limit, and adhering to any proposal type-specific threshold it will be moved to the Governor for voting.
+4. **Governor:** If the `Top100DelegatesProposalValidator`confirms that the proposal is valid, it forwards the proposal to the OP Governor contract. Then, the voting time begins as usual.
+
+### Off-chain flow
+```mermaid
+  flowchart LR
+    Delegate([Delegate])
+    ForumDraft[OPGov Forum]
+    Community([Community])
+    ForumFinal[OPGov Forum]
+
+    Delegate -->|submit DRAFT proposal| ForumDraft
+    Community -->|provide feedback| ForumDraft
+    ForumDraft -->|submit FINAL proposal with proposalId| ForumFinal
+```
+
+### On-chain flow
+```mermaid
+flowchart LR
+    Proposer([Approved Proposer])
+    Validator[ProposalValidator]
+    TopDelegates([Top Delegates])
+    Executor([Executioner])
+    FinalValidator[ProposalValidator]
+    OPGov[OPGov]
+
+    Proposer -->|propose| Validator
+    Validator -->|get proposalId| Proposer
+    TopDelegates -->|approve proposal| Validator
+
+    Executor -->|move to vote| FinalValidator
+    FinalValidator -->|propose| OPGov
+```
+
+# Alternatives Considered
+
+- **Permissioned voting cycle configuration**: As an MVP, we allow a permissioned role to set the voting cycle and delegate approval thresholds. While not fully decentralized, this is acceptable given OF’s existing veto power. A future improvement could use an optimistic oracle to fetch these values on-chain, but this adds complexity we don’t believe is needed yet.
+- **Forum sign-offs with signature verification**: The current process relies on forum comments from top delegates as attestations. We considered maintaining this model, where proposers collect these off-chain signatures and verify them in the `Top100DelegatesProposalValidator`. However, direct on-chain approvals are simpler, reduce errors, and avoid coordination overhead.
```

Comment:
Figuring out exactly how to submit these transactions is one open question we need to address.  Some options I can think of include:
- Chrome extension/bookmarklet to use within the forum
- Standalone local CLI or other tool which pulls data from a forum URL
- Standalone site which pulls data from forum URL
- Ability for proposer to signal proposal finality onchain somehow and allow others to then approve via etherscan or something

Notably we will need to support regular wallets and SAFEs here due to diversity of delegate wallets.

Reply:
Regarding the submission of the proposal, this can either happen through etherscan or Agora's UI.
For the approval of the proposal, by a top delegate, we suggest having this available through Agora's UI where they can display all the proposal data, that the delegate will be called to approve, by listening to an event that is being emitted when a proposal is successfully submitted.